Cannot refer to `spotbugs.toolVersion` from dependencies block on v4 plugin versions

In prior versions of the plugin; I was able to refer to `spotbugs.toolVersion` inside a dependencies block to ensure compile time versions of dependencies are aligned with what the plugin will use internally.

```
dependencies {
    compileOnly "com.github.spotbugs:spotbugs-annotations:${spotbugs.toolVersion}"
    testCompileOnly "com.github.spotbugs:spotbugs-annotations:${spotbugs.toolVersion}"
}
```

This doesn't seem to work any longer (even if I override the `toolVersion`) despite the reference to this property [in the README](https://github.com/spotbugs/spotbugs-gradle-plugin#spotbugs-version-mapping).

The following error occurs:
```
> Task :compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Could not resolve all files for configuration ':compileClasspath'.
   > Could not find com.github.spotbugs:spotbugs-annotations:extension 'spotbugs' property 'toolVersion'.
     Required by:
         project :
```

Workaround is to use fix the version manually like the below; but it seems a bit weird that the old approach doesn't work any longer.

```
ext {
    spotbugsToolVersion = '4.0.1'
}

dependencies {
    spotbugs "com.github.spotbugs:spotbugs:${spotbugsToolVersion}"
    compileOnly "com.github.spotbugs:spotbugs-annotations:${spotbugsToolVersion}"
    testCompileOnly "com.github.spotbugs:spotbugs-annotations:${spotbugsToolVersion}"
}
```
Possibly due to a plugin/extension initialization lifecycle change - if this is intentional and you prefer people to use the `spotbugs` configuration instead, I think that's fine - but would be good to update the `README` to remove references to `spotbugs.toolVersion`.

--- a/README.md
+++ b/README.md
@@ -106,7 +106,7 @@ You can change SpotBugs version by [the `toolVersion` property of the spotbugs e
 
 ### Refer the version in the build script
 
-From v4, the `spotbugs.toolVersion` is changed from `String` to [`Provider<String>`](https://docs.gradle.org/current/javadoc/org/gradle/api/provider/Property.html), so use `get()` or other methods to refer the actual version.
+From v4, the `spotbugs.toolVersion` is changed from `String` to [`Provider<String>`](https://docs.gradle.org/current/javadoc/org/gradle/api/provider/Property.html), so use `get()` or other methods to refer to the actual version.
 
 ```groovy
 dependencies {

--- a/README.md
+++ b/README.md
@@ -104,6 +104,16 @@ You can change SpotBugs version by [the `toolVersion` property of the spotbugs e
 | 4.0.7| 4.0.2|
 | 4.0.0| 4.0.0|
 
+### Refer the version in the build script
+
+From v4, the `spotbugs.toolVersion` is changed from `String` to [`Provider<String>`](https://docs.gradle.org/current/javadoc/org/gradle/api/provider/Property.html), so use `get()` or other methods to refer the actual version.
+
+```groovy
+dependencies {
+    compileOnly "com.github.spotbugs:spotbugs-annotations:${spotbugs.toolVersion.get()}"
+}
+```
+
 ## Copyright
 
 Copyright &copy; 2019-present SpotBugs Team

--- a/src/functionalTest/groovy/com/github/spotbugs/snom/ExtensionFunctionalTest.groovy
+++ b/src/functionalTest/groovy/com/github/spotbugs/snom/ExtensionFunctionalTest.groovy
@@ -239,4 +239,28 @@ spotbugs {
         assertEquals(TaskOutcome.SUCCESS, result.task(":spotbugsMain").outcome)
         assertTrue(result.output.contains("SpotBugs 4.0.0-beta4"))
     }
+
+    def "can use toolVersion to get the SpotBugs version"() {
+        setup:
+        buildFile << """
+spotbugs {
+    toolVersion = "4.0.2"
+}
+dependencies {
+    compileOnly "com.github.spotbugs:spotbugs-annotations:\${spotbugs.toolVersion.get()}"
+}"""
+        when:
+        BuildResult result =
+                GradleRunner.create()
+                .withProjectDir(rootDir)
+                .withArguments(":spotbugsMain", "--debug")
+                .withPluginClasspath()
+                .forwardOutput()
+                .withGradleVersion(version)
+                .build()
+
+        then:
+        result.task(":spotbugsMain").outcome == TaskOutcome.SUCCESS
+        result.output.contains("com.github.spotbugs:spotbugs-annotations:4.0.2")
+    }
 }

